.eslintignore

```diff
@@ -0,0 +1 @@
+test-published-types
```

Comment:
this should be added in eslintrc's `ignorePatterns` instead of using an eslintignore file. it should also be added to package.json's `publishConfig.ignore` list, so we don't ever include it in the tarball.